cluster/addons/rbac/cluster-autoscaler/cluster-autoscaler-rbac.yaml

```diff
@@ -7,10 +7,10 @@ metadata:
 rules:
   # leader election
   - apiGroups: [""]
-    resources: ["endpoints"]
+    resources: ["leases"]
```

Comment:
this should allow both leases and endpoints for at least one release overlap, otherwise loading this manifest will immediately make running autoscaler pods get forbidden errors

also, the `leases` resource is not in the `""` API group, so you'd need:
```yaml
- apiGroups: ["coordination.k8s.io"]
  resources: ["leases"]
  verbs: ["create"]
- apiGroups: ["coordination.k8s.io"]
  resources: ["leases"]
  resourceNames: ["cluster-autoscaler"]
  verbs: ["get", "update", "patch", "delete"]
# TODO: remove N releases after upgrading cluster autoscaler to a version that uses leases
- apiGroups: [""]
  resources: ["endpoints"]
  verbs: ["create"]
- apiGroups: [""]
  resources: ["endpoints"]
  resourceNames: ["cluster-autoscaler"]
  verbs: ["get", "update", "patch", "delete"]
```

Reply:
thanks @liggitt i will make that change.  Should this merge before the autoscaler change or the other way ?

Reply:
this PR should also include a manifest change to bump the version of the cluster autoscaler to the version that starts using leases, right? that means the upstream change should go first, and this PR would bump the autoscaler version along with required permissions

Reply:
@liggitt i updated this PR. I dont see a manifest for cluster auto scaler in the addons folder. 
@aleksandra-malinowska  do you know where the manifest version needs to be bumped  ?

Reply:
huh... if there's no manifest, why do we have an RBAC manifest for it? the RBAC manifest should live alongside the deployment or pod manifest for the autoscaler

Reply:
ah i think this is the one https://github.com/kubernetes/kubernetes/blob/59f0f2d2f9cf24564895a80a7640260a6ff27602/cluster/gce/manifests/cluster-autoscaler.manifest

Reply:
so does this PR need to update the version or invocation in https://github.com/kubernetes/kubernetes/blob/master/cluster/gce/manifests/cluster-autoscaler.manifest as well?

Reply:
I do not believe order matters very much here to be honest. We are typically building first alpha version of CA for given k8s version towards end of release cycle. To follow pattern you suggest we need to either:
 * leave it hanging around for another ~3 months
 * release some extra alpha CA image which uses leases now.

The latter does not directly give any value i believe. And I do not like the former as it is nicer to have things closed that hanging around in the air.

Reply:
I would expect the manifest deploying the autoscaler and the one granting RBAC permissions to be in the same repo. It seems like this is changing permissions for the benefit of an autoscaler manifest living elsewhere? If so, I'd recommend adding the RBAC roles/bindings to that manifest as well, so you don't depend on this one in the kubernetes repo.

When we're ready to update the autoscaler manifest that lives in this repo to a version that actually needs these permissions, then it makes more sense to update this role manifest

Reply:
> When we're ready to update the autoscaler manifest that lives in this repo to a version that actually needs these permissions, then it makes more sense to update this role manifest

Ok - so I guess will wait for a while until we have first alpha version for 1.17 ready.

Reply:
Is this ready to go now ?

Reply:
until this change is needed/used by the version of the cluster autoscaler in the manifest checked into k8s.io/kubernetes, I wouldn't expect a disconnected permission change like this to merge